### ISSUE TITLE
Panic in generate-key-pair when trying to write to a directory without permissions

### ISSUE THREAD

**Issue by Ikke** [sigstore/cosign#2161]
**Description**

When running `cosign generate-key-pair` in a directory where it cannot write to, it results in a panic:

```
$ docker run -it --rm -e COSIGN_PASSWORD= -u1000:1000  gcr.io/projectsigstore/cosign:v1.10.1 generate-key-pair
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xd7bd7c]

goroutine 1 [running]:
github.com/sigstore/cosign/pkg/cosign.FileExists({0x27668bd?, 0x0?})
        github.com/sigstore/cosign/pkg/cosign/common.go:35 +0x5c
github.com/sigstore/cosign/cmd/cosign/cli/generate.GenerateKeyPairCmd({0x2b8bd80, 0xc0001ac000}, {0x0?, 0x0?}, {0x3f38378?, 0x0?, 0x0?})
        github.com/sigstore/cosign/cmd/cosign/cli/generate/generate_key_pair.go:88 +0x22a
github.com/sigstore/cosign/cmd/cosign/cli.GenerateKeyPair.func1(0xc0006be000?, {0x3f38378?, 0x0?, 0x0?})
        github.com/sigstore/cosign/cmd/cosign/cli/generate_key_pair.go:66 +0x4c
github.com/spf13/cobra.(*Command).execute(0xc0006be000, {0x3f38378, 0x0, 0x0})
        github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc00017a000)
        github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
        github.com/sigstore/cosign/cmd/cosign/main.go:61 +0x4c
```

**Version**

1.10.1

**Comment by znewman01** [comment]
This is closed by #2162 right?

**Comment by znewman01** [comment]
Closing as fixed by #2162.
